packages/material-ui/src/Tabs/Tabs.js

```diff
@@ -2,96 +2,217 @@ import * as React from 'react';
     flexDirection: 'column',
+  }),
+  ...(styleProps.scrollButtonsHideMobile && {
+    [`.${tabsClasses.scrollButtons}`]: {
+      [theme.breakpoints.down('sm')]: {
+        display: 'none',
+      },
+    },
+  }),
+}));
+
+const TabsScroller = experimentalStyled(
+  'div',
+  {},
+  {
+    name: 'MuiTabs',
+    slot: 'Scroller',
+    overridesResolver: (props, styles) => {
+      const { styleProps } = props;
+      return deepmerge(
+        {
+          ...(styleProps.fixed && styles.fixed),
+          ...(styleProps.hideScrollbar && styles.hideScrollbar),
+          ...(styleProps.scrollableX && styles.scrollableX),
+          ...(styleProps.scrollableY && styles.scrollableY),
+        },
+        styles.scroller || {},
+      );
+    },
   },
-  /* Styles applied to the flex container element if `centered={true}` & `!variant="scrollable"`. */
-  centered: {
-    justifyContent: 'center',
-  },
-  /* Styles applied to the tablist element. */
-  scroller: {
-    position: 'relative',
-    display: 'inline-block',
-    flex: '1 1 auto',
-    whiteSpace: 'nowrap',
-  },
-  /* Styles applied to the tablist element if `!variant="scrollable"`. */
-  fixed: {
+)(({ styleProps }) => ({
+  position: 'relative',
+  display: 'inline-block',
+  flex: '1 1 auto',
+  whiteSpace: 'nowrap',
+  ...(styleProps.fixed && {
     overflowX: 'hidden',
     width: '100%',
-  },
-  /* Styles applied to the tablist element if `variant="scrollable"` and `orientation="horizontal"`. */
-  scrollableX: {
-    overflowX: 'auto',
-    overflowY: 'hidden',
-  },
-  /* Styles applied to the tablist element if `variant="scrollable"` and `orientation="vertical"`. */
-  scrollableY: {
-    overflowY: 'auto',
-    overflowX: 'hidden',
-  },
-  /* Styles applied to the tablist element if `variant="scrollable"` and `visibleScrollbar={false}`. */
-  hideScrollbar: {
+  }),
+  ...(styleProps.hideScrollbar && {
     // Hide dimensionless scrollbar on MacOS
     scrollbarWidth: 'none', // Firefox
     '&::-webkit-scrollbar': {
       display: 'none', // Safari + Chrome
     },
-  },
-  /* Styles applied to the ScrollButtonComponent component. */
-  scrollButtons: {},
-  /* Styles applied to the ScrollButtonComponent component if `allowScrollButtonsMobile={true}`. */
-  scrollButtonsHideMobile: {
-    [theme.breakpoints.down('sm')]: {
-      display: 'none',
+  }),
+  ...(styleProps.scrollableX && {
+    overflowX: 'auto',
+    overflowY: 'hidden',
+  }),
+  ...(styleProps.scrollableY && {
+    overflowY: 'auto',
+    overflowX: 'hidden',
+  }),
+}));
+
+const FlexContainer = experimentalStyled(
+  'div',
+  {},
+  {
+    name: 'MuiTabs',
+    slot: 'FlexContainer',
+    overridesResolver: (props, styles) => {
+      const { styleProps } = props;
+      return deepmerge(
+        {
+          ...(styleProps.flexContainerVertical && styles.flexContainerVertical),
+          ...(styleProps.centered && styles.centered),
+        },
+        styles.flexContainer || {},
+      );
```

Comment:
```suggestion
      return {
        ...styles.flexContainer,
        ...(styleProps.vertical && styles.flexContainerVertical),
        ...(styleProps.centered && styles.centered),
      };  
```

Let's remove `flexContainerVertical` from styleProps, no need for introducing additional prop.